libs/experimental/langchain_experimental/graph_transformers/llm.py

```diff
@@ -99,6 +157,91 @@ class _Graph(BaseModel):
     relationships: Optional[List]
 
 
+class UnstructuredRelation(BaseModel):
+    head: str = Field(
+        description=(
+            "extracted head entity like Microsoft, Apple, John. "
+            "Must use human-readable unique identifier."
+        )
+    )
+    head_type: str = Field(
+        description="type of the extracted head entity like Person, Company, etc"
+    )
+    relation: str = Field(description="relation between the head and the tail entities")
+    tail: str = Field(
+        description=(
+            "extracted tail entity like Microsoft, Apple, John. "
+            "Must use human-readable unique identifier."
+        )
+    )
+    tail_type: str = Field(
+        description="type of the extracted tail entity like Person, Company, etc"
+    )
+
+
+def create_unstructured_prompt(
+    node_labels: Optional[List[str]] = None, rel_types: Optional[List[str]] = None
+) -> ChatPromptTemplate:
+    system_prompt = f"""You are a top-tier algorithm designed for extracting 
+information in structured formats to build a knowledge graph. Your task is 
+to identify the entities and relations requested with the user prompt, from 
+a given text. You must generate the output in a JSON containing a list with 
+JSON objects having the following keys: "head", "head_type", "relation", 
+"tail", and "tail_type". The "head" key must contain the text of the extracted 
+entity with one of the types from the provided list in the user prompt. 
+The "head_type" key must contain the type of the extracted head entity which 
+must be one of the types from {node_labels}. The "relation" key must contain 
+the type of relation between the "head" and the "tail" which must be one of 
+the relations from {rel_types}. The "tail" key must represent the text of an 
+extracted entity which is the tail of the relation, and the "tail_type" key 
+must contain the type of the tail entity from {node_labels}. Attempt to 
+extract as many entities and relations as you can. Maintain Entity 
+Consistency: When extracting entities, it's vital to ensure consistency.
+If an entity, such as "John Doe", is mentioned multiple times in the text 
+but is referred to by different names or pronouns (e.g., "Joe", "he"), 
+always use the most complete identifier for that entity. The knowledge graph 
+should be coherent and easily understandable, so maintaining consistency in 
+entity references is crucial.
+
+IMPORTANT NOTES:
+- Don't add any explanation and text."""
+
+    system_message = SystemMessage(content=system_prompt)
+    parser = JsonOutputParser(pydantic_object=UnstructuredRelation)
+
+    human_prompt = PromptTemplate(
+        template="""Based on the following example, extract entities and 
+relations from the provided text.\n\n
+Use the following entity types, don't use other entity that is not defined below:
```

Comment:
should we add this conditionally to the prompt depending if we have node_labels and rel_types?

Reply:
sure